pkg/subctl/cmd/unexport.go

```diff
@@ -0,0 +1,85 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/cmd/utils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	mcsclient "sigs.k8s.io/mcs-api/pkg/client/clientset/versioned/typed/apis/v1alpha1"
+)
+
+var (
+	unexportCmd = &cobra.Command{
+		Use:   "unexport",
+		Short: "Stop a resource from being exported to other clusters",
+		Long:  "This command stops exporting a resource so it isn't accessible to other clusters any more",
+	}
+	unexportServiceCmd = &cobra.Command{
+		Use:   "service <serviceName>",
+		Short: "Stop a Service from being exported to other clusters",
+		Long: "This command removes the ServiceExport resource with the given name which in turn stops the Service " +
+			"of the same name from being exported to other clusters",
+		PreRunE: restConfigProducer.CheckVersionMismatch,
+		Run:     unexportService,
+	}
+)
+
+func init() {
+	restConfigProducer.AddKubeContextFlag(unexportCmd)
+	addNamespaceFlag(unexportServiceCmd)
+	unexportCmd.AddCommand(unexportServiceCmd)
+	rootCmd.AddCommand(unexportCmd)
+}
+
+func unexportService(cmd *cobra.Command, args []string) {
+	err := validateUnexportArguments(args)
+	utils.ExitOnError("Insufficient arguments", err)
+
+	clientConfig := restConfigProducer.ClientConfig()
+	restConfig, err := clientConfig.ClientConfig()
+	utils.ExitOnError("Error connecting to the target cluster", err)
+
+	client, err := mcsclient.NewForConfig(restConfig)
+	utils.ExitOnError("Error connecting to the target cluster", err)
+
+	if namespace == "" {
+		if namespace, _, err = clientConfig.Namespace(); err != nil {
+			namespace = "default"
+		}
+	}
+
+	err = client.ServiceExports(namespace).Delete(context.TODO(), args[0], metav1.DeleteOptions{})
```

Comment:
We should ignore IsNotFound() error in case export is already deleted. Or return a non-error msg that ServiceExport was not found. In most unexport cases users won't care if it was already deleted but throwing error may cause problems for their cleanup scripts.

Reply:
I was following `kubectl` in this regard, as it does throw an error when attempting to delete a resource which isn't found.

In case of scripts an error can be easily avioded.

If we really want to simplify we can add `--ignore-not-found[=false]` flag as exists in `kubectl delete` [1] but I think it's not necessary at this point and can always be added in the future.

[1] https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#delete